.github/workflows/build.yaml

```
@@ -29,13 +31,25 @@ jobs:
     runs-on: "${{matrix.os}}"
     steps:
       - uses: "actions/checkout@v4"
-
-      - run: "cmake -B ./build -DCMAKE_BUILD_TYPE=${{matrix.build_type}}"
-
-      - run: "cmake --build ./build --config ${{matrix.build_type}}"
-
-      - run: "cmake --install ./build --prefix ./install"
-
-      - run: "cmake -S examples -B ./examples/build -DCMAKE_BUILD_TYPE=${{matrix.build_type}}"
-
-      - run: "cmake --build ./examples/build --config ${{matrix.build_type}}"
+        with:
+          submodules: "recursive"
+
+      - name: "Install Catch2 on macOS"
+        if: "matrix.os == 'macos-latest'"
+        run: "brew install catch2"
+
+      - name: "Install Catch2 on Ubuntu"
+        if: "matrix.os == 'ubuntu-latest'"
+        working-directory: "tools/deps-install/ubuntu"
+        run: "./install-catch2.sh 3.6.0"
+
+      - name: "Build Executables"
+        run: |
+          cmake -B ./build -DCMAKE_BUILD_TYPE=${{matrix.build_type}}
+          cmake --build ./build --config ${{matrix.build_type}}
+          cmake --install ./build --prefix ./install
+          cmake -S examples -B ./examples/build -DCMAKE_BUILD_TYPE=${{matrix.build_type}}
+          cmake --build ./examples/build --config ${{matrix.build_type}}
+
+      - name: "Run Unit Tests"
+        run: "./build/tests/unit-test"
```

```suggestion
        run: "cmake --build ./build --target test"
```

This should work after making the suggested cmake changes.

---

Used ctest instead